index.js

```diff
@@ -52,8 +57,8 @@ function fastifyCors (fastify, opts, next) {
   next()
 }
 
-function handleCorsOptionsDelegator (optionsResolver, fastify) {
-  fastify.addHook('onRequest', function onRequestCors (req, reply, next) {
+function handleCorsOptionsDelegator (optionsResolver, fastify, { hookName } = { hookName: defaultHookName }) {
+  fastify.addHook(hookName, function handleCors (req, reply, next) {
```

Comment:
Were do we define the variable hookName?

Reply:
it's being passed as parameter and defaults to `defaultHookName`

Reply:
I dont see a `const hookName=` in your new code?!